File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/dialog/FirstTimeDialogActivity.java
package com.dansoftware.libraryapp.gui.firsttime.dialog;

import com.dansoftware.libraryapp.appdata.Preferences;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class FirstTimeDialogActivity {

    private final Preferences preferences;

    public FirstTimeDialogActivity(@NotNull Preferences preferences) {
        this.preferences = Objects.requireNonNull(preferences, "Preferences shouldn't be null");
    }

    public void show() {
        FirstTimeDialogWindow window = new FirstTimeDialogWindow(new FirstTimeDialog(preferences));
        window.showAndWait();
    }

}

File: src/main/java/com/dansoftware/libraryapp/gui/firsttime/dialog/FirstTimeDialogActivity.kt
package com.dansoftware.libraryapp.gui.firsttime.dialog

import com.dansoftware.libraryapp.appdata.Preferences
import java.util.*

/**
 * Used for showing [FirstTimeDialog] with a [FirstTimeDialogWindow].
 *
 * @author Daniel Gyorffy
 */
class FirstTimeDialogActivity(private val preferences: Preferences) {
    init {
        Objects.requireNonNull(preferences, "Preferences shouldn't be null")
    }

    fun show() {
        val window = FirstTimeDialogWindow(FirstTimeDialog(preferences))
        window.showAndWait()
    }
}